components/script/dom/htmlanchorelement.rs

```diff
@@ -657,8 +658,24 @@ pub fn follow_hyperlink(subject: &Element, hyperlink_suffix: Option<String>) {
             .get_attribute_by_name(DOMString::from_string(String::from("referrerpolicy")))
             .and_then(|attribute: DomRoot<Attr>| (determine_policy_for_token(&attribute.Value())));
 
-        // Step 13, 14.
+        // Step 13
+        let noreferrer =
+            if let Some(link_types) = subject.get_attribute(&ns!(), &local_name!("rel")) {
+                let values = link_types.Value();
+                let contains_noreferrer = values.contains("noreferrer");
+
+                contains_noreferrer
+            } else {
+                false
+            };
+        let referrer = if noreferrer {
```

Comment:
Have you considered collapsing the above two conditionals into one `let referrer =`? I think you might also be able to replace the if/else with chaining various methods of `Option`.

Reply:
what do you think of the collapsed version?

https://github.com/servo/servo/pull/23090/commits/0808e700e6bc008386830763d98def7b009f33eb#diff-2a94daab4b979b5338e64bef3081e92cR662